### ISSUE TITLE
Sticky effect: problem when combine with menu/datatable and dialog

### ISSUE THREAD

**Issue by uvula** [primefaces/primefaces#12151]
### Describe the bug

A view is combine menu, datatable with sticky effect and also having a dialog, the "z-index" is not handled correct.
Required steps are described in the Reproducer pages itself.

Checked with firefox/chrome/edge. Same problem

### Reproducer

Reproducer added

### Expected behavior

I'd like to have menu on top
Datatable is configured to bounce below the menu.
A dialog is below the datatable header and not visible at all until doing some scrolling/mouse wheel

### PrimeFaces edition

Community

### PrimeFaces version

14.0.1

### Theme

Saga

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

**Comment by uvula** [comment]
[primefaces-test.zip](https://github.com/user-attachments/files/15905455/primefaces-test.zip)

**Comment by melloware** [comment]
You reproducer is using 13.0.7 not 14.0.1 ?

**Comment by melloware** [comment]
[pf-12151-14.zip](https://github.com/user-attachments/files/15905630/pf-12151-14.zip)

I updated your demo to 14.0.1 and i can't reproduce your issue.

**Comment by uvula** [comment]
[primefaces-test.zip](https://github.com/user-attachments/files/15905638/primefaces-test.zip)
Sorry, sorry, Eclipse/Git imported in another directory, was not in eclipse workspace. I zipped old case. Attached again.

**Comment by melloware** [comment]
I can't reproduce.  I followed the instructions and it works for me on 14.0.1

**Comment by melloware** [comment]
![sticky](https://github.com/primefaces/primefaces/assets/4399574/6bfb33eb-3fd9-4669-bc72-c981e95f8eb0)

**Comment by uvula** [comment]
Sorry, was uploading wrong zip from old issue. I uploaded again. Please check new ZIP. Sorry for confusing!

**Comment by melloware** [comment]
try this version:
[pf-12151.zip](https://github.com/user-attachments/files/15906062/pf-12151.zip)

**Comment by uvula** [comment]
![sticky-dialog-behind](https://github.com/primefaces/primefaces/assets/1205197/a398f92b-f14f-4885-8029-508a6ca83b2c)
![sticky-menu-behind](https://github.com/primefaces/primefaces/assets/1205197/dafaca7c-72ea-4d63-a413-5915486ccb4c)

**Comment by melloware** [comment]
are you sure you tried my latest version attached?

**Comment by melloware** [comment]
![sticky](https://github.com/primefaces/primefaces/assets/4399574/ac04903f-df1e-4dc4-a9af-97bc7faf5181)

**Comment by uvula** [comment]
Sorry second time, I didn't expect your are so fast. I just added the screenshot to make it clear for your investiagtion. Late here in Germany in the meantime :-) I thought 12151 was based on my wrong loaded old issue.
Pulled pf-12151.zip and tested.
Cleaning browser cache.

Enter view:
1.) Menu is above the datatable header, also during scrolling -> OK
2.) Dialog is above the datatable header -> OK
Looks perfect.
Thanks a lot

**Comment by uvula** [comment]
In my existing project (PF14.0.1), I added the test.js fix and tested too. I guess I found an additional situation with some problem on the z-index. I try to add to reproducer and send new zip.

**Comment by uvula** [comment]
I copied node.xhtml and added an accordion to the node-and-accordion.xhtml view. Accordion is also behind datatable header, when starting.
Just one scroll will fix. Maybe fix is easy, if not -> user should just scroll once with mouse wheel
[pf-12151.zip](https://github.com/user-attachments/files/15912078/pf-12151.zip)

**Comment by uvula** [comment]
Problem 4: why problem 4? Just to ask, is my approach correct?
I can't bring the problem to pf-12151 reproducer, I tried, but I can't "break it".
My approach: I added test.js to **MY APPLICATION** into main.xhtml -> last loading of the fix to override the function nextStickyZindex
  (<h:head> ->  <h:outputScript name="test.js" />)
I see test.js is loaded (browser debug)
In my app, there is one long menu, overlap the datatable header, it is behind :-(. I have no idea. 
Maybe when moving to 14.0.2 is will work, I guess not.
With all the last fixes in "sticky", the UI is now much better, also Problem4, just one scroll operation, and it is fine now.

<img width="577" alt="menu-behind" src="https://github.com/primefaces/primefaces/assets/1205197/c83272d1-d482-4e82-9544-e5fd15b6d726">

**Comment by melloware** [comment]
I don;t think you uploaded the right code as your node-with accordion does not look like above and everything is working for me.

**Comment by uvula** [comment]
??? I downloaded second pf-12151.zip back and I see my node-and-accordion.xthml is here (and also the changed NodeView.java)
It looks:
<img width="110" alt="ac01" src="https://github.com/primefaces/primefaces/assets/1205197/2f801959-195a-4a2d-a0de-73978e18f689">
<img width="787" alt="ac02" src="https://github.com/primefaces/primefaces/assets/1205197/a8ca5d21-f709-4768-b412-6978ffb5c1c2">
<img width="1262" alt="ac03" src="https://github.com/primefaces/primefaces/assets/1205197/0deca2ad-56d0-414a-b84f-129354ccf220">

Problem4: it looks in my application, but I can't bring the problem to the reproducer, just explaining, what I have done here (test.js etc). Bring the behavior to reproducer will not break actually. Again: but with all the fixes yet, just one mouse wheel operation and the menu is correct.

**Comment by melloware** [comment]
OK I don't think we will fix that as accordion does not contain a z-index and nor should it.    It looks like the fix you needed for https://github.com/primefaces/primefaces/issues/11988 is what is causing it.  If I remove that then you won't be able to filter if I add it back you get this Accordion issue.

Basically my advice is "you are abusing the use of Sticky" and you should consider changing your UI.    Even having two sticky's on a page seems like a UI/UX smell for me.

**Comment by melloware** [comment]
@uvula your complex scenario causes too many issues.  I am reverting the fix as reported in #12293 

You have a monkeypatch you can use your for use case but two sticky's does not seem like a common scenario etc.

**Comment by melloware** [comment]
@uvula here is a new Monkeypatch i think it still solves your scenario and the new one.

```js
PrimeFaces.utils.nextStickyZindex = function() {
  var highestStickyZIndex = parseInt($('.ui-sticky:visible').last().zIndex()) || parseInt(PrimeFaces.nextZindex()) + 1;

  var overlays = $('.ui-widget-overlay:visible');
  if (overlays.length) {
    overlays.each(function() {
      var overlayZIndex = parseInt($(this).zIndex()) - 1;
      highestStickyZIndex = Math.min(overlayZIndex, highestStickyZIndex);
    });
  } else {
    PrimeFaces.zindex = highestStickyZIndex - 1;
  }

  return highestStickyZIndex - 1;
};
```
